include/certifier_framework.h

```diff
@@ -202,18 +225,64 @@ namespace certifier {
       bool save_store();
       bool fetch_store();
       void clear_sensitive_data();
-      bool cold_init(const string& public_key_alg, const string& symmetric_key_alg);
+
+      bool generate_symmetric_key(bool regen);
+      bool generate_sealing_key(bool regen);
+      bool generate_auth_key(bool regen);
+      bool generate_service_key(bool regen);
+
+      bool cold_init(const string& public_key_alg, const string& symmetric_key_alg,
+                     int asn1_cert_size, byte* asn1_cert,
+                     const string& home_domain_name, const string& home_host,
+                     int home_port, const string& service_host, int service_port);
       bool warm_restart();
-      bool certify_me(const string& host_name, int port);
-      bool recertify_me(const string& host_name, int port, bool generate_new_key);
       bool GetPlatformSaysAttestClaim(signed_claim_message* scm);
       void print_trust_data();
 
+      bool certify_primary_domain();
+      bool certify_me() {return certify_primary_domain();};
+
       // For peer-to-peer certification (not used yet)
       bool init_peer_certification_data(const string& public_key_alg);
       bool recover_peer_certification_data();
       bool get_peer_certification(const string& host_name, int port);
       bool run_peer_certification_service(const string& host_name, int port);
+
+      // multi-domain support
+      bool add_new_domain(const string& domain_name, const string& cert, const string& host, int port,
+          const string& service_host, int service_port);
+      bool certify_secondary_domain(const string& domain_name);
+      bool get_certifiers_from_store();
+      bool put_certifiers_in_store();
+    };
+
+    // Certification Anchors
+
+    class certifiers {
```

Comment:
Because we use the singular of certifier in so many ways, I decided not to overload it with another thing.  Of course, we could change it to something else.